diff --git a/tests/functional/local-overlay-store/stale-file-handle-inner.sh b/tests/functional/local-overlay-store/stale-file-handle-inner.sh
index 46f5eb2..2c42d98 100755
--- a/tests/functional/local-overlay-store/stale-file-handle-inner.sh
+++ b/tests/functional/local-overlay-store/stale-file-handle-inner.sh
@@ -33,8 +33,12 @@ triggerStaleFileHandle () {
     buildInStore "$storeB"
 }
 
-# Without remounting, we should encounter errors
-expectStderr 1 triggerStaleFileHandle | grepQuiet 'Stale file handle'
+# Without remounting, we should encounter errors. However, this doesn't seem to
+# happen on Linux 6.19+ anymore.
+#
+# See https://github.com/NixOS/nixpkgs/issues/496466
+( expectStderr 1 triggerStaleFileHandle | grepQuiet 'Stale file handle' ) || \
+    skipTest "Couldn't trigger the error"
 
 # Configure remount-hook and reset OverlayFS
 storeB="$storeB&remount-hook=$PWD/remount.sh"
